pandas/_libs/tslibs/strptime.pyx

```diff
@@ -118,6 +121,7 @@ def array_strptime(ndarray[object] values, object fmt, bint exact=True, errors='
     result = np.empty(n, dtype='M8[ns]')
     iresult = result.view('i8')
     result_timezone = np.empty(n, dtype='object')
+    new_result = np.empty(n, dtype='O8')
```

Comment:
also i am a bit concerned that this is going to be very heavyweight (e.g. this is going to be 10x memory as an object array is much heavier here). so even though this is a bit more complicated, i would put the main loop in a try/except and then if it excepts then do the object dtype conversion (only). now I don't entirely remember this code so this may blow up complexity but it is worth thinking about.

this pattern we do elsewhere (though @jbrockmendel maybe we removed / reduced it?)

Reply:
yah thats the pattern we use in array_to_datetime

also why 'O8' instead of just 'O'?

Reply:
Do you suggest that is better to make the main loop twice if an exception was thrown than having a second array for the main loop lifetime? Okay, that can be possible. I will check `array_to_datetime` and see how did it was done.

Reply:
yep exactly

Reply:
In respect to "O" or "O8", I thought I needed to maintain the dtype size, if before was an "I8" and transformed to object type it had to be in 8 bytes too.

Reply:
O is just fine (or `dtype=object`

Reply:
I just added the new function just like u said. Now I created `array_strptime_object` that makes the same as `array_strptime` but returns the array with object dtype. I ran all the tests and all seems to work fine.
I did exactly the same as the function u gave to me. `array_to_datetime_object`